utils/fix_rules.py

```diff
@@ -257,6 +286,77 @@ def rewrite_value_remove_prefix(line):
     return key + ": " + new_value
 
 
+class CCEFile:
+    def __init__(self, project_root=None):
+        if not project_root:
+            project_root = os.path.join(
+                os.path.dirname(os.path.abspath(__file__)), "..")
+        self.project_root = project_root
+
+    @property
+    def absolute_path(self):
+        raise NotImplementedError()
+
+    def line_to_cce(self, line):
+        return line
+
+    def line_isnt_cce(self, cce, line):
+        return line != cce
+
+    def read_cces(self):
+        with open(self.absolute_path, "r") as f:
+            cces = f.read().splitlines()
+        return cces
```

Comment:
Does it make sense to filter for valid CCEs and only return CCE-like entries?

What's the format of this file? If it is strictly one CCE per line, do we need to do the line 311? Could we instead make this class closeable and do something like:

```python3
with CCEFile() as picker:
   ... assign cces ... 
``` 

And have the file be read once (into this class) and cces removed on pick and the file re-written from `.close()`? It means you probably have to use `random.randint` instead of `random.shuffle` over all CCEs, but might be cleaner.

My 2c.

Reply:
It's not that simple, `with` calls the close by default even when there is an exception, which would delete a CCE from the file even though the code that is supposed to write it into the rule fails. Also the current architecture also requires to come up with a potential CCE - that may not end up being assigned at all.
So this explicit approach ask for CCE + ask for its deletion when we know that it has been used looks best to me under current conditions, although I understand that it is not ideal.